src/pcost.py

```diff
@@ -0,0 +1,31 @@
+import csv
+from pathlib import Path
+from typing import Union
+
+
+def portfolio_cost(file_path: Union[str, Path]) -> float:
+    """
+    Calculates the cost of the portfolio by using the amount of shares
+    and the corresponding price.
+
+    :param file_path: Path to the portfolio file
+    :return: The cost of the portfolio
+    """
+    total_cost = 0.0
+    with open(file_path) as f:
+        rows = csv.reader(f)
+        headers = next(rows)
+        for rowno, row in enumerate(rows, 1):
+            if not row:
+                continue
+            rowdict = dict(zip(headers, row))
+            try:
+                shares = int(rowdict["shares"])
+                price = float(rowdict["price"])
+                total_cost += shares * price
+
+            # Exercise asks to catch the error like this
+            except ValueError:
+                print(f"Row {rowno}: Bad row {rowdict}")
```

Comment:
Instead of the `print` we should use [logging](https://docs.python.org/3/library/logging.html) package:
```suggestion
                logging.info(f"Row {rowno}: Bad row {rowdict}")
```

For complex applications it allows to have more control on how we display information to the user